src/libs/actions/IOU.js

```diff
@@ -1464,53 +1465,61 @@ function payMoneyRequest(paymentType, chatReport, iouReport) {
  * Initialize money request info and navigate to the MoneyRequest page
  * @param {String} iouType
  * @param {String} reportID
+ * @returns {Promise}
  */
 function startMoneyRequest(iouType, reportID = '') {
-    resetMoneyRequestInfo(`${iouType}${reportID}`);
-    Navigation.navigate(ROUTES.getMoneyRequestRoute(iouType, reportID));
+    return resetMoneyRequestInfo(`${iouType}${reportID}`).then(() => {
+        Navigation.navigate(ROUTES.getMoneyRequestRoute(iouType, reportID));
+    });
```

Comment:
It's pretty uncommon to have to wait for a merge to navigate. If we are going to chain off of `Onyx.merge()` I think we should leave some kind of comment to explain why this is the exception to the rule.

Reply:
I think the problem here is that now when we have withOnyx improvement we may sometimes read outdate value. For instance we have a screen that operates on value "X". Let say we had a flow that first modifies the value "X" then navigates to our screen. Before what was happening is withOnyx wrapped screen was waiting for onyx values by waiting for connect method and then info about value "X". in connect method there were a lot of promises so we were sure that our modification was succeeded before we actually rendered content of the new screen. Now it's possible that we will modify the "X" navigate to screen read previous value of "X" from cache then once modification takes place we render the component one more time.

Reply:
waiting for merge prevents reading outdate value from what I understand

Reply:
I think we can easily miss some cases as previously we didn't need to care about it as we were sure that we won't read from onyx before the modification happen. 
Maybe we should persist what is stale and don't read from cache when it is or indicate that we don't worry about reading outdated data.

Reply:
I would see  it this way. 
If we don't mind reading outdated data (want to show messages as soon as possible we can add a flag)
```JS
withOnyx({
  allowReadingStaleData: true
})
```

Then on every merge/set/remove updates two simple maps so we know if the data in cache is up to date
StaleStorageLastOperation[key] = operation_number_that_modifies_the_key
toDoOperations.push(operationId)
Then
isDataStale = (key) => (toDoOperation.has(StaleStorageLastOperation[key])).

if the data is stale we don't get it from cache in withOnyx wrapper unless someone set `allowReadingStaleData ` to true